modules/phase_field/src/postprocessors/FeatureFloodCount.C

```diff
@@ -579,20 +562,19 @@ FeatureFloodCount::mergeSets(bool use_periodic_boundary_info)
   } // map loop
 
 
-  for (unsigned int map_num = 0; map_num < _maps_size; ++map_num)
+  for (auto map_num = decltype(_maps_size)(0); map_num < _maps_size; ++map_num)
     _feature_sets[map_num].clear();
 
   // Now consolidate the data structure
   _feature_count = 0;
-  for (processor_id_type rank = 0; rank < n_procs; ++rank)
-    for (unsigned int map_num = 0; map_num < _maps_size; ++map_num)
+  for (auto rank = decltype(_n_procs)(0); rank < _n_procs; ++rank)
+    for (auto map_num = decltype(_maps_size)(0); map_num < _maps_size; ++map_num)
     {
-      for (std::vector<FeatureData>::iterator it = _partial_feature_sets[rank][map_num].begin();
-           it != _partial_feature_sets[rank][map_num].end(); ++it)
+      for (auto & feature : _partial_feature_sets[rank][map_num])
       {
-        if (!it->_merged)
+        if (!feature._merged)
         {
-          _feature_sets[map_num].push_back(MooseSharedPointer<FeatureData>(new FeatureData(*it)));
+          _feature_sets[map_num].emplace_back(std::unique_ptr<FeatureData>(new FeatureData(feature)));
```

Comment:
Right, I read about that in Meyer's book. I commented on the libMesh issue to have John or Roy add make_unique wrapper to libMesh so we don't have to wait several years to start using that feature.